src/databases/PLOT3D/vtkPLOT3DReader.C

```diff
@@ -790,20 +855,85 @@ vtkPLOT3DReader::ReadGrid(FILE *xyzFp)
   this->SkipByteCount(xyzFp);
   int d = this->Internal->NumberOfDimensions;
   if (this->ReadVector(xyzFp, this->NumberOfPoints, d, pointArray) == 0)
-    {
+  {
     vtkErrorMacro("Encountered premature end-of-file while reading "
                   "the geometry file (or the file is corrupt).");
     return VTK_ERROR;
+  }
+
+  // Read IBlanking data
+  std::cout << "IblankingInFile: " << this->IBlankingInFile << std::endl;
+  std::cout << "Use IBlanking If Detected: " << this->UseIBlankingIfDetected << std::endl;
+  if (this->IBlankingInFile == 1 && this->UseIBlankingIfDetected == 1)
+  // if (this->Internal->IBlanking == 1)
+  {
+    vtkIntArray* iblank = vtkIntArray::New();
+    iblank->SetNumberOfTuples(this->NumberOfPoints);
+
+    // Attempt to read the iblank data from the plot3d file. If the read failes,
+    // then the function will return 0 and we will error out. If the read passes,
+    // it will return the count that indicates how many numbers it read.
+    if (this->ReadIBlank(xyzFp, this->NumberOfPoints, iblank) == 0)
+    {
+      vtkErrorMacro("Encountered premature end-of-file while reading "
+                    "the iblanking data from the xyz file.");
+      return VTK_ERROR;
+    }    
+
+    // IBlank data is 0 if it should be blanked and 1 if it should not be
+    // blanked, which is a little backwards from how I think of it, but okay.
+    // So where ib is 0 we use NODE_NOT_APPLICABLE_TO_PROBLEM and where ib is
+    // 1 we use DUPLICATED_NODE.
+    vtkUnsignedCharArray *ghostNodes = vtkUnsignedCharArray::New();
+    ghostNodes->SetName("avtGhostNodes");
+    ghostNodes->SetNumberOfTuples(this->NumberOfPoints);
+    ghostNodes->FillComponent(0, 0);
+    int *ib = iblank->GetPointer(0);
+    unsigned char *gnp = ghostNodes->GetPointer(0);
+    for (int i = 0; i < this->NumberOfPoints; ++i)
+    {
+      if (ib[i] == 0)
+      {
+        avtGhostData::AddGhostNodeType(gnp[i], avtGhostNodeTypes::NODE_NOT_APPLICABLE_TO_PROBLEM);
+      }
+      else if (ib[i] == 1)
+      {
+        gnp[i] = 0;
+      }
     }
 
-  // This is where we would read IBlanking information, but VisIt currently 
-  //  doesn't do anything with it
-  // 
-#if 0
-  if (this->Internal->IBlanking)
+    output->GetPointData()->AddArray(ghostNodes);
+    iblank->Delete();
+    ghostNodes->Delete();
+
+    // Based on the iblanked nodes, determine which zones should be iblanked.
+    vtkUnsignedCharArray *ghostZones = vtkUnsignedCharArray::New();
+    ghostZones->SetNumberOfValues(output->GetNumberOfCells());
+    ghostZones->SetName("avtGhostZones");
+    vtkIdList* ids = vtkIdList::New();
+    ids->SetNumberOfIds(8);
```

Comment:
Setting the number of Ids is unnecessary here. The call to GetCellPoints will set it.